### DISCUSSION TITLE
alpha and beta chain dimers from mixcr

### DISCUSSION THREAD

**Discussion by malonzm1** [milaboratory/mixcr#1443] [Q&A] [answered]
Hi,

I've used mixcr with many samples from a huge scRNAseq dataset. In each case, for TCRs, I only get either alpha or beta chain in a cell and never both as a dimer. Any thoughts?

Thanks and good day.

Br,

**Comment by mizraelson** [comment]
Hi,

It's impossible to provide a definitive answer without knowing specific details about your experimental protocol, the MiXCR commands you've used, the reports you've obtained, and without a closer examination of the raw data. The effectiveness and outcomes of T-cell receptor (TCR) analysis in single-cell RNA sequencing (scRNA-seq) data can vary significantly based on various factors.

To better understand and address the issue you're experiencing, please provide more information about the following:

1) Experimental Protocol: Describe the scRNA-seq protocol you used, including any details about cell isolation, library preparation, and sequencing technologies.

2) MiXCR Commands: Share the specific MiXCR commands or parameters you employed for TCR analysis.

3) Reports: Provide any relevant reports or output generated by MiXCR during your analysis.

4) Raw Data: If possible, share a sample of the raw data or data processing steps to help identify potential issues.

Best regards

**Reply by malonzm1** [reply]
Thanks.

1. I ran mixcr with various samples (GSE114374, GSE116222, GSE132771, GSE133028, GSE134649, GSE134809, GSE138669, GSE140881, GSE142016, GSE142637, GSE148020, GSE148837, GSE149133, GSE151177, GSE153765, GSE154775, GSE156310, GSE156776, GSE157278, GSE157477, GSE159117, GSE160097, GSE161500, GSE163121, GSE163314, GSE165645, GSE175990, GSE178833, GSE179640, GSE181957, GSE193770, GSE198891, GSE203262, GSE208161, GSE213216, GSE215237, GSE138266, GSE147424, GSE169147, GSE180759, GSE148073, GSE137029) with various experimental protocols.

2. One MiXCR command is: `mixcr analyze 10x-sc-5gex -s hsa --rna --threads 4 --force-overwrite %s %s %s/%s'%(infile1,infile2,outdir,srr)`, where infile1 contains barcodes and infile2 coding reads.

3. One report is in the attached file
[mixcr_GSE114374.txt](https://github.com/milaboratory/mixcr/files/13486352/mixcr_GSE114374.txt)

4. I used fastq files from NCBI GEO directly (or converted from bam files from NCBI GEO).

Thanks and good day.

**Reply by mizraelson** [reply]
Are any of your datasets a targeted 10x 5' VDJ library? For non target 5' GEX data its hard to guarantee the presence of both chains for cells as most of the reads do not cover VDJ region.

**Reply by malonzm1** [reply]
A few are. Like GSE125527. 
[mixcr2_GSE125527_1.txt](https://github.com/milaboratory/mixcr/files/13493567/mixcr2_GSE125527_1.txt)

**Reply by mizraelson** [reply]
According to the report you have shared there was an error and the run failed. Can you share the output clonotype table for this sample?

**Reply by malonzm1** [reply]
Below are samples of clones.tsv for this study.
[Uploading GSE125527_0.tar.gz…]()

**Reply by mizraelson** [reply]
Does the link work for you? I cant access the file.

**Reply by malonzm1** [reply]
[GSE125527_0.tar.gz](https://github.com/milaboratory/mixcr/files/13494152/GSE125527_0.tar.gz)

**Reply by malonzm1** [reply]
The more recent attachment works for me.

**Reply by mizraelson** [reply, marked as answer]
According to the [description](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE125527) ,  all the files you've shared are from 5' GEX, not VDJ. In the study, they conducted both analyses, and for each donor, there are three types of samples: scRNA, BCR, and TCR (e.g., C9_R_10x_scRNA, C9_R_10x_BCR, C9_R_10x_TCR).

When analyzing a BCR sample like this one:
```
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR847/000/SRR8478650/SRR8478650_1.fastq.gz
ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR847/000/SRR8478650/SRR8478650_2.fastq.gz
```
and running the MiXCR command:

```
mixcr  analyze 10x-sc-xcr-vdj \
--species hsa \
SRR8478650_GSM3576432_C9_R_10x_BCR_Homo_sapiens_OTHER_{{R}}.fastq.gz \
output
```

I found 1297 cells with both heavy and light chains identified. There was also minor T cell contamination, with 15 T cells detected, including 5 with both TRA and TRB chains. Please see the attached output files for more details.
[output.clone.groups_TRAB.txt](https://github.com/milaboratory/mixcr/files/13509254/output.clone.groups_TRAB.txt)
[output.clone.groups_IG.txt](https://github.com/milaboratory/mixcr/files/13509255/output.clone.groups_IG.txt)

**Reply by malonzm1** [reply]
Many thanks! I noticed you shared output.clone.groups_*.txt and not output.clones.tsv. Sorry if this is an ignorant question, may I know the difference between the two? And what are cellGroups? 

From what I understand, output.clone.groups.txt is a relatively new feature as my earlier MiXCR runs only generated output.clones.tsv (not output.clone.groups.tsv). I compared the two files for a sample I ran with MiXCR recently (SRR11561460) and found dimers in output.clone.groups.tsv but not in output.clones.tsv. Would you recommend I re-run analyses without output.clone.groups.tsv?

**Reply by mizraelson** [reply]
The file output.clone.groups.txt is generated by the [mixcr exportCloneGroups](https://mixcr.com/mixcr/reference/mixcr-export/#clone-groups-by-cell) command in MiXCR. In recent version, MiXCR has enhanced its ability to segregate clone groups by cells, specifically excluding cells that contain mixed chains, such as IG and TCR within a single cell. All clones listed in this file are also included in the standard output.clones.tsv. However, the grouped clones format in output.clone.groups.txt simplifies the process of analyzing cells. You can find more detailed information in the [MiXCR documentation](https://mixcr.com/mixcr/reference/mixcr-groupClones/), and the topic is also discussed here #1400.

**Reply by malonzm1** [reply]
Many thanks. Sorry if this has an obvious answer. In output.clone.groups.tsv there are column names with TRA.primary.*, TRA.second.*, TRB.primary.*, TRB.second.*, etc. What is the difference between primary and second?

**Reply by mizraelson** [reply]
`Primary` and `second` are the two most abundant clones in the cell. In the latest MiXCR for 10x preset the abundance is calculated by UMI count."

**Reply by malonzm1** [reply]
Thanks! Is there any way to determine the cell barcode of the cellGroup in output.clone.groups.tsv? Is it correct that each cellGroup has a unique cell barcode?

**Reply by mizraelson** [reply]
`cellGroup` represent not a single cell, but a unique pair of chains (clone). Thus there is no information on the cell barcode, but rather `groupUniqueCellCount` column contains the number of cells (cell barcodes) for a particular clone (pair of chains).

**Reply by malonzm1** [reply]
Thanks. Is it correct that all cells in a cellGroup have the clone (unique pair of chains)? Is there a way to identify which cells (e.g. via cell barcode) are included in a cellGroup?

**Reply by mizraelson** [reply]
Yes, all cells in the `cellGroup` bear the same set of receptor chains. In the non grouped output table you can see `cellGroup` value for every cell which corresponds to the groups in the grouped output. Please use the latest [develop version](https://link.milaboratory.com/software/mixcr/mixcr-develop.zip).

**Reply by malonzm1** [reply]
Thanks. What are "undefined" cell groups? And why can the same cell groups have different barcodes?

**Reply by mizraelson** [reply]
In some cases, it's not possible to assign a cell to any group; in such cases, the group is labeled as `undefined`. By definition, a cell group is a collection of cells (not a single cell), which is why there are multiple cell barcodes comprising a group. Imagine there are 100 cells, each bearing the same pair of chains. Then there are 100 different cell barcodes, but they are grouped together based on the identical clone sequence into one group.

**Comment by malonzm1** [comment]
Thanks. Can I still use the clone information from *.clones.tsv from a cell even if its cellGroup is undefined?

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by mizraelson** [comment]
Yes, you can use the clone information. But in some cases you can't be sure about the alpha-beta chain pairing in that cell.
